interrupts.h

```diff
@@ -0,0 +1,35 @@
+#ifndef __INTERRUPTS_H__
+#define __INTERRUPTS_H__
+
+#include "pic32mz.h"
+#include "global_config.h"
+
+/* Disables interrupts. Returns the previous value of Status
+   register. */
+#define di() ({                             \
```

Comment:
FYI: FreeBSD declares two helper functions to disable preemption:

    void critical_enter();
    void critical_exit();